### DISCUSSION TITLE
Nested properties obtained from `getValues` are erased on unregister

### DISCUSSION THREAD

**Discussion by malinin01** [react-hook-form/react-hook-form#7709] [Q&A] [answered]
### Version Number

7.25.3

### Codesandbox/Expo snack

https://codesandbox.io/s/pedantic-sun-9qkk0?file=/src/App.js

### Steps to reproduce

1. On the main page
2. fill in the name input
3. fill in the surname input
4. tick membership checkbox 
5. open console in the devtools
6. click submit button
7. look at the last parameter/object of _"render submitted values:"_
8. the object is empty

### Expected behaviour

The object should contain name and surname properties.

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
render submitted values:  {personal: Object, isMember: true} true {}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment, marked as answer]
If you're manually unregistering your input fields via the [`unregister` form method](https://react-hook-form.com/api/useform/unregister/#main) and want your values to be kept in RHF state, you should set the `unregister` option `keepValue: true`
```js
unregister("isMember", { keepValue: true });
```
or not call the `unregister` method at all.

Refer to [the `unregister` section](https://react-hook-form.com/api/useform/unregister/#main) of RHF docs for more info
See [this CSB](https://codesandbox.io/s/async-dream-cfd3f?file=/src/App.js) for a working example

**Comment by malinin01** [comment]
**More context:**
I have already seen some discussions about immutability in this library and saw a general disposition towards not deeply cloning the result of `getValues` within this library. What surprises me, the repro above works with v7.0.1 without cloning the result of `getValues`. Hence disregarding immutability, I would expect such change to be handled as a breaking release and avoid implicitly broken apps considering the popularity of this library.

**Comment by malinin01** [comment]
Thank you for the answer @Moshyfawn! btw the `isMember` field is not an issue. It's persisted after unregistering even without the `keepValue` option thanks to the shallow copy inside `getValues`. However, as you have hinted setting `{keepValue: true}` on the other fields indeed fixes the reported issue with the name and surname inputs (apparently, this is why repro works with v7.0.1), although it's unintended to keep these fields in internal RHF state. 

Nevertheless, returning a deep clone from `getValues` sounds like a more sustainable solution to unbind consumers from the internal state of RHF, although being less performant.

**Comment by bluebill1049** [comment]
This is expected behavior when invoking `unreigster` method.
